### DISCUSSION TITLE
so I have a Dockerfile from add_dockerfile_shinyproxy() ... what's next?

### DISCUSSION THREAD

**Discussion by nickcardamone** [ThinkR-open/golem#569] [Q&A]
Hi,

New to Docker and Shinyproxy so this question could be due to a lack of knowledge on both of those fronts, but is there any documentation on how to construct a directory with the golem-created Dockerfile that can be interpreted by ShinyProxy? 

My goal is to deploy my shiny app which is packaged using golem via Docker and Shinyproxy and the only documentation I'm seeing is without golem. 

E.g. https://medium.com/analytics-vidhya/deploying-a-shiny-flexdashboard-with-docker-cca338a10d12

And https://towardsdatascience.com/an-open-source-solution-to-deploy-enterprise-level-r-shiny-applications-2e19d950ff35

I also saw Colin Fay's helpful article, but he also doesn't use golem: https://rtask.thinkr.fr/shiny-application-in-production-with-shinyproxy-docker-and-debian/

**Comment by ColinFay** [comment]
Hey @ncardamo, 

I'm converting this issue to a discussion as I think it's not a bug or dev related but more about finding information on a specific topic :) 

Now that you have your container ready and build locally, here are the steps to follow.

- Have Shiny Proxy installed on your server. You can follow the steps at [rtask.thinkr.fr/shiny-application-in-production-with-shinyproxy-docker-and-debian/](https://rtask.thinkr.fr/shiny-application-in-production-with-shinyproxy-docker-and-debian/) or the ones from their own website [shinyproxy.io/getting-started/](https://www.shinyproxy.io/getting-started/), or any other from the internet. As long as you have the hello world app running, that's good :) 

- Send the folder containing the application to your server where you will be running Shiny Proxy. I suggest doing this through git (git commit on your local machine, push to your git server, then git clone on the server with SP). 

- Connect to your remote server, go to the folder where the golem app is, and build the container. For example 

```
cd mygolemapp && docker build -t myappimage .
```

That will create a docker image called `myappimage` on your server, you should see it listed when doing `docker images`. This `myappimage` will be reused in `container-image` in the SP yaml.

- Verify that you can launch the container by running `docker run myappimage`. This should print the shiny logs to the console.

- Modify your Shiny Proxy YAML to incorporate the name of the docker image you have just built, for example with 

```yaml
  - id: my_golem_app
    container-cmd: ["R", "-e", "mygolemapp::run_app()"]
    container-image: myappimage
    container-network: sp-example-net
```

- Restart SP 

- Enjoy 🎊

The reason why there is no documentation so far on SP + `{golem}` is that `{golem}` based container don't have anything special compared to the other containers: they perform the same task (using a function that launches an app), the only difference being the way `{golem}` based app are built. 

Hope this helps, 

Colin

**Reply by nickcardamone** [reply]
Thanks a lot for this write-out Colin.

I'll try this out and report back.

Nick
